### ISSUE TITLE
Released v2.0 on Feb 2, 2020

### ISSUE THREAD

**Issue by x448** [fxamacker/cbor#117]
See [release notes](https://github.com/fxamacker/cbor/releases/tag/v2.0.0) for more info.

## Changes in v2.0:
* Improved API.
* Faster performance.
* Reduced memory usage.
* Improved code quality using around 20 linters.
* Replaced EncOptions.TimeRFC3339 bool with TimeMode (5 settings).
* Decode NaN and Infinity time values to Go's "zero time" value.
* Removed deprecated v1 options and Valid() function.

CBOR Tags (major type 6) was moved to milestone v2.1. and is over 50% done.

## Improved API
In v2, more function signatures match encoding/json. These are the most widely used:
`Marshal`, `Unmarshal`, `NewEncoder`, `NewDecoder`, `encoder.Encode`, `decoder.Decode`.

"Mode" in this API means definite way of encoding or decoding.

EncMode and DecMode are interfaces created from EncOptions or DecOptions structs.

EncMode and DecMode use immutable options so their behavior won't change at runtime.  Modes are intended to be reused and are safe for concurrent use.

```
// Create EncOptions, using either struct literal or a function.
opts := cbor.CanonicalEncOptions()

// If needed, modify options -- e.g. how time values should be encoded.
opts.Time = cbor.TimeUnix

// Create reusable EncMode interface with immutable options, safe for concurrent use.
em, err := opts.EncMode()   

// Use EncMode like encoding/json, with same function signatures.
b, err := em.Marshal(v)
// or
encoder := em.NewEncoder(w)
err := encoder.Encode(v)
```

__Package Level Functions__

These package level functions use default options without CBOR tags. Their API matches encoding/json.  EncMode and DecMode also provide these functions with same signatures.
```
b, err := cbor.Marshal(v)
err := cbor.Unmarshal(b, &v)
encoder := cbor.NewEncoder(w)
decoder := cbor.NewDecoder(r)
```

__EncOptions.Time__

Replaced EncOptions.TimeRFC3339 bool with TimeMode:
* TimeUnix                   // secs, converted to smallest CBOR integer that fits
* TimeUnixMicro           // μs, CBOR float with subsecs in fractional part
* TimeUnixDynamic      // secs or μs, either int or float depending on empty subsecs
* TimeRFC3339            // secs, string
* TimeRFC3339Nano    // ns, string with trailing zeros removed

**Comment by fxamacker** [comment]
v2.0 just passed 1 billion execs using Go 1.12.  :tada:  It will be released tonight.

**Comment by fxamacker** [comment]
Hi @kostko @rumpelsepp @yawning 

I released v2.0 and would like to release v2.1 on Feb 9 with CBOR tags support.

Would you be interested in providing feedback on v2.1 API if I post an issue with the draft API?

__Why 2.0?__

v2.0 was needed to simplify adding new features planned for v2.1 (CBOR tags) and v2.2.

__Roadmap__
* v2.1 (Feb. 9, 2020) support for CBOR tags (major type 6) and some decoder optimizations.
* v2.2 (Feb. 2020) options for handling duplicate map keys.

CBOR Tags (major type 6) was moved to milestone v2.1. and is over 50% done.

__Status__

v2.0 passed 1+ billion execs in coverage-guided fuzzing before release.  This is the most well-tested and most reliable release of this library.

See [releases](https://github.com/fxamacker/cbor/releases) for more info about v2.0, v1.5, etc.

**Comment by kostko** [comment]
> I released v2.0 and would like to release v2.1 on Feb 9 with CBOR tags support.

Nice! :-)

> Would you be interested in providing feedback on v2.1 API if I post an issue with the draft API?

Sure, feel free to cc me in the issue with the new API. I am especially interested in how the type registration will look like. The way it was handled in go-codec was slightly annoying.

**Comment by fxamacker** [comment]
Closing but feel free to comment.
